workflow/controller/operator.go

```diff
@@ -817,6 +814,11 @@ func (woc *wfOperationCtx) checkReconciliationComplete() bool {
 	return woc.wf.Status.Phase.Completed() && !woc.wf.Status.TaskResultsInProgress()
 }
 
+func (woc *wfOperationCtx) checkTaskResultsComplete() bool {
+	woc.log.Debugf("Task results completion status: %v", woc.wf.Status.TaskResultsCompletionStatus)
+	return !woc.wf.Status.TaskResultsInProgress()
+}
```

Comment:
I was always a little reluctant to call this `checkTaskResultsComplete()` since it could imply that all TaskResults that would ever exist are complete. So, my preference was to say `checkTaskResultsInProgress()` and flip the boolean. Would it be okay to do that?